src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,153 @@
+package duke;
+
+import java.time.format.DateTimeFormatter;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+/**
+ * Class containing methods for UI-related functionality.
+ */
+public class Ui {
+    private static final String SEPARATOR =
+            "____________________________________________________________";
+    private static final String PREFIX = "\t";
+    private static final String LOGO = " ____        _        \n"
+            + "|  _ \\ _   _| | _____ \n"
+            + "| | | | | | | |/ / _ \\\n"
+            + "| |_| | |_| |   <  __/\n"
+            + "|____/ \\__,_|_|\\_\\___|\n";
+    private static final String WELCOME_MSG =
+            "Hello from\n" + LOGO + "What can I do for you?";
+    private static final String BYE_MSG = "Bye. Hope to see you again soon!";
+
+    /**
+     * Renders a task into a string.
+     * @param task Task.
+     * @return Input task rendered as a string.
+     */
+    public static String renderTask(Task task) {
+        var taskType = TaskType.identifyTask(task);
```

Comment:
Interesting use of `var`. :) (and elsewhere in your code)
To give another perspective of this however, without an ide, say I'm viewing the codebase on github, I think it would be harder to tell what is the type of the variable. Hence, I'm not so sure if this is the "Java" way. I suppose this is still way better than reading raw code from many dynamically typed languages out there.
Regardless, viewing code without an ide probably isn't a major usecase these days so this shouldn't be an issue.

Reply:
I'm not sure if `var` is permissible nowadays and the points you mentioned are certainly valid. I don't think it's a coding standard violation and I still can't find any standards that recommend for/against it, so I figured I'd just use it whenever I can to avoid having to put the types in 😂 Let me know if you find out more!